lib/Frontend/CompilerInstance.cpp

```diff
@@ -2014,6 +2014,12 @@ CompilerInstance::loadModule(SourceLocation ImportLoc,
                      Module, ImportLoc);
   }
 
+  // Resolve any remaining module using export_as for this one.
+  getPreprocessor()
+      .getHeaderSearchInfo()
+      .getModuleMap()
+      .resolveLinkAsDependencies(TopModule);
```

Comment:
This is already called in ASTReader. Does it need to be called again?

Reply:
Yes. It's very common that we build a module and then read it back in the same compiler invocation. This would load the module again with the flag off (since we don't serialize it). This is also the reason I'm using names to keep track instead of 'Module *', since I wanna be sure we are always looking at the most recent load.